index.js

```diff
@@ -1,61 +1,186 @@
+'use strict';
+
 const core = require('@actions/core');
 const github = require('@actions/github');
 const http = require('@actions/http-client');
 
-async function main() {
-    let sha = github.context.sha;
+function GitlabClient (inputConfig, githubConfig) {
+    this.inputConfig = inputConfig;
+    this.githubConfig = githubConfig;
+    this.client = new http.HttpClient();
 
-    switch (github.context.eventName) {
-        case "pull_request":
-        case "pull_request_target":
-            sha = github.context.payload.pull_request.head.sha;
-        case "push":
-            break;
-        case "schedule":
-            break;
-
-        default:
-            core.debug('Not a push event, bailing out');
-
-            return;
-    }
+    this.request = async function (method, url, data) {
+        let res;
+
+        url.searchParams.append('access_token', this.inputConfig.accesToken);
+
+        if (method === 'GET') {
+            res = await this.client.get(url);
+        } else if (method === 'POST') {
+            const additionalHeaders = {
+                'Content-Type': 'application/json'
+            };
+            res = await this.client.post(url, data, additionalHeaders);
+        } else {
+            throw new Error(`Unsupported method: ${method}`);
+        }
+
+        if (res.message.statusCode !== 200 && res.message.statusCode !== 201) {
+            throw new Error(`API request [${method}] ${url.pathname} failed with code ${res.message.statusCode}`);
+        }
+
+        const body = await res.readBody();
+        return JSON.parse(body);
+    };
+
+    this.get = async function (url) {
+        return await this.request('GET', url);
+    };
+
+    this.post = async function (url, data) {
+        return await this.request('POST', url, data);
+    };
+
+    this.fetchRunningPipelinesIds = async function () {
+        const twoHoursBefore = new Date();
+        twoHoursBefore.setHours(twoHoursBefore.getHours() - 2);
+
+        const url = new URL(`projects/${this.inputConfig.projectId}/pipelines`, this.inputConfig.apiUrl);
+        url.searchParams.append('ref', this.inputConfig.triggeredRef);
+        url.searchParams.append('status', 'running');
+        url.searchParams.append('per_page', '100');
+        url.searchParams.append('updated_after', twoHoursBefore.toISOString());
```

Comment:
Do we need this if we're also filtering for running pipelines?

Reply:
Yes, I think so. There are some forever running pipelines, probably because of a gitlab bug. Hence there is no point in cancelling them every time, because they cannot be cancelled anyway. But now I'm thinking about those 2 hours. It might be a bit too little. But 4 should be plenty IMO